src/nvim/testdir/test_system.vim

```diff
@@ -12,7 +12,9 @@ function! Test_System()
 
   let out = systemlist('echo 123')
   " On Windows we may get a trailing space and CR.
-  if out != ["123 \r"]
+  if &shell =~# 'cmd.exe$'
+    call assert_equal(['123\r'], out)
+  elseif out != ["123 \r"]
```

Comment:
why not just `else`?

Reply:
The deleted line didn't include `has('win32')` so I thought it may happen in other OS.